internal/provider/resource_gitlab_personal_access_tokens.go

```diff
@@ -0,0 +1,265 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var validPersonalAccessTokenScopes = []string{
+	"api",
+	"read_user",
+	"read_api",
+	"read_repository",
+	"write_repository",
+	"read_registry",
+	"write_registry",
+	"sudo",
+}
+
+var _ = registerResource("gitlab_personal_access_token", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_personal_access_token`" + ` resource allows to manage the lifecycle of a personal access token for a specified user.
+
+-> **Note**: In order to utilize this resource without failure the executing user must be an admin. 
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/personal_access_tokens.html)
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/users.html#create-a-personal-access-token)`,
```

Comment:
It would be ok to only include the first link (https://docs.gitlab.com/ee/api/personal_access_tokens.html) here, since that page links to the other one.

By the way, try out the preview took here: https://registry.terraform.io/tools/doc-preview . Without a new line between each paragraph, you get the text on the same line.

Reply:
Will do! Thank you for that I had no idea that tool existed.  I usually use the right click preview in VS code but for some reason it is broken in chrome on a GitPod and results in a 401.